src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2750,6 +2754,14 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, JVMFlagOrigin
         return JNI_ERR;
 #endif // INCLUDE_MANAGEMENT
 #if INCLUDE_JVMCI
+    } else if (match_option(option, "-XX:+EnableJVMCI")) {
+        jio_fprintf(defaultStream::error_stream(),
+                  "-XX:+EnableJVMCI was removed in JDK 25. Use --add-modules=jdk.internal.vm.ci to make use of the JVMCI module.\n");
+        return JNI_EINVAL;
+    } else if (match_option(option, "-XX:-EnableJVMCI")) {
+        jio_fprintf(defaultStream::error_stream(),
+                  "-XX:-EnableJVMCI was removed in JDK 25.\n");
+        return JNI_EINVAL;
```

Comment:
Since `EnableJVMCI` is an experimental flag, it should be removed by the "2 step" model. It should be entered in `special_jvm_flags[]` in arguments.cpp as obsoleted in JDK 25.

https://github.com/openjdk/jdk/blob/7bde2bb57159aaac36a6a585f70c4672919c8c16/src/hotspot/share/runtime/arguments.cpp#L483-L484